Carbon.Domain.EntityFrameworkCore.Extensions/Repositories/TenantManagedEFRepositoryBase.cs

```diff
@@ -31,45 +31,44 @@ public TenantManagedEFRepositoryBase(DbContext ctx)
 
         public override async Task ConnectToSolution<T>(T relatedEntity)
         {
-            if (relatedEntity == null || relatedEntity.RelationalOwners == null || !relatedEntity.RelationalOwners.Any() || relatedEntity.RelationalOwners.Where(k => k == null).Any())
-                return;
-            bool relationsChanged = false;
-            
-            foreach (var ro in relatedEntity.RelationalOwners)
+			if (relatedEntity == null || relatedEntity.RelationalOwners.Where(k => k == null).Any())
+				return;
+
+			if (relatedEntity.RelationalOwners == null)
             {
-                ro.EntityId = relatedEntity.Id;
-                ro.EntityCode = relatedEntity.GetObjectTypeCode();
+				relatedEntity.RelationalOwners = new List<EntitySolutionRelation>();
             }
 
-            var oldRelations = await TargetErDbSet.Where(k => !k.IsDeleted && k.EntityCode == relatedEntity.GetObjectTypeCode() && k.EntityId == relatedEntity.Id).ToListAsync();
+			foreach (var ro in relatedEntity.RelationalOwners)
+			{
+				ro.EntityId = relatedEntity.Id;
+				ro.EntityCode = relatedEntity.GetObjectTypeCode();
+			}
 
-            if (oldRelations == null || !oldRelations.Any())
-                relationsChanged = true;
-            else
-            {
-                var firstSet = new HashSet<Guid>(oldRelations.Select(k => k.SolutionId).ToList());
-                var secondSet = new HashSet<Guid>(relatedEntity.RelationalOwners.Select(k => k.SolutionId).ToList());
+			var oldRelations = await TargetErDbSet.Where(k => !k.IsDeleted && k.EntityCode == relatedEntity.GetObjectTypeCode() && k.EntityId == relatedEntity.Id).ToListAsync();
 
-                var allEqual = secondSet.SetEquals(firstSet);
-                if (!allEqual)
-                {
-                    relationsChanged = true;
```

Comment:
buradaki allEqual ile başlayan if bloğunun içinde relationsChanged=true yapıldığından buraya giren bir code path mecburen aşağıdaki (62. satır) if bloğuna da giriyor, bu yüzden kodun yeni versiyonunda bu iki if bloğu birleştirildi. bu şekilde iki ayrı yerde DB için SaveChangesAsync çağırılmasının da önüne geçilmiş oluyor.